x-pack/plugins/apm/public/components/app/service_overview/service_overview_instances_table/instance_actions_menu/menu_sections.ts

```diff
@@ -0,0 +1,203 @@
+
+  return {
+    from: timeInMilliseconds - fiveMinutes,
+    to: timeInMilliseconds + fiveMinutes,
+  };
+}
+
+export function getMenuSections({
+  instanceDetails,
+  basePath,
+  onFilterByInstanceClick,
+  analizeRuntimeMetricsHref,
+}: {
+  instanceDetails: InstaceDetails;
+  basePath: IBasePath;
+  onFilterByInstanceClick: () => void;
+  analizeRuntimeMetricsHref: string;
+}) {
+  const podId = instanceDetails.kubernetes?.pod?.uid;
+  const containerId = instanceDetails.container?.id;
+  const time = instanceDetails['@timestamp']
+    ? new Date(instanceDetails['@timestamp']).valueOf()
+    : undefined;
+  const infraMetricsQuery = getInfraMetricsQuery(instanceDetails['@timestamp']);
+
+  const podActions: Action[] = [
+    {
+      key: 'podLogs',
+      label: i18n.translate(
+        'xpack.apm.serviceOverview.instancesTable.actionMenus.podLogs',
+        { defaultMessage: 'Pod logs' }
+      ),
+      href: getInfraHref({
+        app: 'logs',
+        basePath,
+        path: `/link-to/pod-logs/${podId}`,
+        query: { time },
+      }),
+      condition: !!podId,
+    },
+    {
+      key: 'podMetrics',
+      label: i18n.translate(
+        'xpack.apm.serviceOverview.instancesTable.actionMenus.podMetrics',
+        { defaultMessage: 'Pod metrics' }
+      ),
+      href: getInfraHref({
+        app: 'metrics',
+        basePath,
+        path: `/link-to/pod-detail/${podId}`,
+        query: infraMetricsQuery,
+      }),
+      condition: !!podId,
+    },
+  ];
+
+  const containerActions: Action[] = [
+    {
+      key: 'containerLogs',
+      label: i18n.translate(
+        'xpack.apm.serviceOverview.instancesTable.actionMenus.containerLogs',
+        { defaultMessage: 'Container logs' }
+      ),
+      href: getInfraHref({
+        app: 'logs',
+        basePath,
+        path: `/link-to/container-logs/${containerId}`,
+        query: { time },
+      }),
+      condition: !!containerId,
+    },
+    {
+      key: 'containerMetrics',
+      label: i18n.translate(
+        'xpack.apm.serviceOverview.instancesTable.actionMenus.containerMetrics',
+        { defaultMessage: 'Container metrics' }
+      ),
+      href: getInfraHref({
+        app: 'metrics',
+        basePath,
+        path: `/link-to/container-detail/${containerId}`,
+        query: infraMetricsQuery,
+      }),
+      condition: !!containerId,
+    },
+  ];
+
+  const apmActions: Action[] = [
+    {
+      key: 'filterByInstance',
+      label: i18n.translate(
+        'xpack.apm.serviceOverview.instancesTable.actionMenus.filterByInstance',
+        {
+          defaultMessage: 'Filter overview by instance',
+        }
+      ),
+      onClick: onFilterByInstanceClick,
+      condition: true,
+    },
+    {
+      key: 'analyzeRuntimeMetric',
+      label: i18n.translate(
+        'xpack.apm.serviceOverview.instancesTable.actionMenus.analyzeRuntimeMetric',
+        {
+          defaultMessage: 'Analyze runtime metric',
```

Comment:
Is this supposed to be "metrics"? I've never heard this phrase and didn't know what this link did until I clicked it.

Reply:
@smith do you mean changing from `Analyze runtime metric` to `metrics` to `Analyze runtime metrics`?

Reply:
Definitely should be "metrics" 👍